egs/how2/st1/run.sh

```diff
@@ -0,0 +1,273 @@
+#!/bin/bash
+
+# Copyright 2019 Kyoto University (Hirofumi Inaguma)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+. ./path.sh || exit 1;
+. ./cmd.sh || exit 1;
+
+# general configuration
+backend=pytorch # chainer or pytorch
+stage=0         # start from -1 if you need to start from data download
+stop_stage=100
+ngpu=1          # number of gpus ("0" uses cpu, otherwise use gpu)
+debugmode=1
+dumpdir=dump    # directory to dump full features
+N=0             # number of minibatches to be used (mainly for debugging). "0" uses all minibatches.
+verbose=0       # verbose option
+resume=         # Resume the training from snapshot
+seed=1          # seed to generate random number
+# feature configuration
+do_delta=false
+
+train_config=conf/train.yaml
+decode_config=conf/decode.yaml
+
+# decoding parameter
+recog_model=model.acc.best # set a model to be used for decoding: 'model.acc.best' or 'model.loss.best'
+
+# model average realted (only for transformer)
+n_average=5                  # the number of ST models to be averaged
+use_valbest_average=true     # if true, the validation `n_average`-best ST models will be averaged.
+                             # if false, the last `n_average` ST models will be averaged.
+
+# pre-training related
+asr_model=
+mt_model=
+
+# preprocessing related
+case=tc
+# tc: truecase
+# lc: lowercase
+# lc.rm: lowercase with punctuation removal
+
+# Set this to somewhere where you want to put your data, or where
+# someone else has already put it.  You'll want to change this
+# if you're not on the CLSP grid.
+how2=/export/a13/kduh/mtdata/how2/how2-300h-v1
+# how2=/home/inaguma/how2/how2-300h-v1
+# how2-300h-v1
+#  |_ data/
+#  |_ features/
+#    |_ fbank_pitch_181516/
+
+# bpemode (unigram or bpe)
+nbpe=8000
+bpemode=bpe
+
+# exp tag
+tag="" # tag for managing experiments.
+
+. utils/parse_options.sh || exit 1;
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+train_set=train.pt
+train_dev=val.pt
+recog_set="dev5.pt"
+
+if [ ${stage} -le 0 ] && [ ${stop_stage} -ge 0 ]; then
+    ### Task dependent. You have to make data the following preparation part by yourself.
+    ### But you can utilize Kaldi recipes in most cases
+    echo "stage 0: Data Preparation"
+    local/data_prep.sh ${how2}
+fi
+
+feat_tr_dir=${dumpdir}/${train_set}/delta${do_delta}; mkdir -p ${feat_tr_dir}
+feat_dt_dir=${dumpdir}/${train_dev}/delta${do_delta}; mkdir -p ${feat_dt_dir}
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    ### Task dependent. You have to design training and dev sets by yourself.
+    ### But you can utilize Kaldi recipes in most cases
+    echo "stage 1: Feature Generation"
+
+    # The fbank features is generated as pre-processing; 40-dimensional fbanks with 3-dimensional pitch on each frame
```

Comment:
Why don't you include the feature extraction process?
I want `run.sh` to be self-consistent.

Reply:
How2 corpus provides pre-processed fbank features (40-dim fbank+3-dim pitch) only, which means we have no chance to access to the original audio files.

Reply:
Thanks!
Now I got it.
Could you add such comments more explicitly in `run.sh`?
Then, it's perfect.